src/fides/api/ops/api/v1/endpoints/user_endpoints.py

```diff
@@ -231,6 +232,18 @@ def update_managed_systems(
     """
     user = validate_user_id(db, user_id)
 
+    if not (user.permissions and user.permissions.roles):  # type: ignore
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail=f"User {user_id} needs permissions before they can be assigned as system manager.",
+        )
+
+    if APPROVER in user.permissions.roles:  # type: ignore
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail=f"User {user_id} is an {APPROVER} and cannot be assigned as a system manager.",
+        )
```

Comment:
It feels like there's a mismatch between the UI and the desired logic on the backend. 

In the UI, you can currently add systems to a user _before_ their role is actually saved.  My backend changes here require a role is saved before systems are added, and would need a corresponding UI change if we went this route.
Because the latest role may not have been saved, the backend from the "Add Systems" window may not have current knowledge of the user's intended role. 


![Screenshot 2023-03-21 at 4 28 08 PM](https://user-images.githubusercontent.com/9755598/226745512-2bcb0289-3495-4a80-bb82-43b575445452.png)


Another scenario that causes issues, say I am a root user and am trying to upgrade an Approver to "Viewer and Approver".  I can't add systems before clicking "Save" on the Viewer and Approver Role, because the user is still an Approver as far as the backend is concerned.

![Screenshot 2023-03-21 at 3 02 29 PM](https://user-images.githubusercontent.com/9755598/226746665-39be1e42-b559-4abd-a9a6-ca5e1c417553.png)

Reply:
FE ticket here: https://github.com/ethyca/fidesplus/issues/722